src/test/java/com/mael/payroll/paymentTypes/HourlyPaymentTest.java

```diff
@@ -0,0 +1,75 @@
+package com.mael.payroll.paymentTypes;
+
+import com.mael.payroll.cards.TimeCard;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.time.LocalDate;
+
+import static java.time.LocalDate.of;
+import static java.time.Month.JANUARY;
+import static org.junit.Assert.assertEquals;
+
+public class HourlyPaymentTest {
+
+    private HourlyPayment hourlyPayment;
+    private LocalDate friday;
+
+    @Before
+    public void setUp() {
+        hourlyPayment = new HourlyPayment(1000.0);
+        friday = of(2016, JANUARY, 29);
+    }
+
+    @Test
+    public void thePayIsZeroIfNoTimeCard() {
+        assertEquals(0.0, hourlyPayment.calculatePay(friday), 0.001);
+    }
+
+    @Test
+    public void thePayIsHoursByRateIfOneTimeCard() {
+        hourlyPayment.addTimeCard(new TimeCard(friday, 8.0));
+        assertEquals(8.0 * 1000.0, hourlyPayment.calculatePay(friday), 0.001);
+    }
+
+    @Test
+    public void thePayIsTheSumOfTotalHoursByRateIfTwoTimeCards() {
+        LocalDate monday = of(2016, JANUARY, 25);
+        hourlyPayment.addTimeCard(new TimeCard(monday, 8.0));
+
+        LocalDate tuesday = of(2016, JANUARY, 26);
+        hourlyPayment.addTimeCard(new TimeCard(tuesday, 8.0));
+
+        assertEquals(8.0 * 1000.0 + 8.0 * 1000.0, hourlyPayment.calculatePay(friday), 0.001);
```

Comment:
I am totally aware of that, but the problem is that I make my branches from the old ones, because I have to continue working while I wait for feedback. As a result, all my branches are chained. 

That also explains why when I delete a parent branch, if I have an open PR for the immediate child branch, it is automatically closed. 

If I added the corrections on top of each PR and then rebased, I would spend most of my time solving merge conflicts. It's ok for one or two branches, as would be the case in a real feature of a real codebase, but not in this case where all my branches are chained and I have loads of them. See [here](https://github.com/me-stevens/payroll/pull/2#issuecomment-169298147).